File: src/main/java/controller/JuegoController.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package controller;

import java.util.ArrayList;
import model.Personaje;

/**
 *
 * @author Usuario
 */
public class JuegoController {
    private ArrayList<Personaje> listaPersonajes;
    public JuegoController(){
        this.listaPersonajes=new ArrayList<>();
    }

    public ArrayList<Personaje> getListaPersonajes() {
        return listaPersonajes;
    }

    public void setListaPersonajes(ArrayList<Personaje> listaPersonajes) {
        this.listaPersonajes = listaPersonajes;
    }
    
    
    //controladores
    
   public void agregarPersonaje(Personaje nuevoPersonaje){
      this.listaPersonajes.add(nuevoPersonaje);
      String sql = "INSERT INTO personajes (nombre, nivel, vida, tipo, atributo_especial) VALUES (?, ?, ?, ?, ?)";
        
        try (java.sql.Connection conexion = db.Conexion.conectar();
             java.sql.PreparedStatement pstmt = conexion.prepareStatement(sql)) {
            pstmt.setString(1, nuevoPersonaje.getNombre());
            pstmt.setInt(2, nuevoPersonaje.getNivel());
            pstmt.setDouble(3, nuevoPersonaje.getVida());
                       if (nuevoPersonaje instanceof model.Guerrero) {
                pstmt.setString(4, "Guerrero");
                pstmt.setDouble(5, ((model.Guerrero) nuevoPersonaje).getAtaqueEspada());
            } else if (nuevoPersonaje instanceof model.Mago) {
                pstmt.setString(4, "Mago");
                pstmt.setDouble(5, ((model.Mago) nuevoPersonaje).getHechizoVida());
            } else if (nuevoPersonaje instanceof model.Arquero) {
                pstmt.setString(4, "Arquero");
                pstmt.setDouble(5, ((model.Arquero) nuevoPersonaje).getAtaqueFlechaVeneno());
            }
            
            pstmt.executeUpdate();
            System.out.println("Personaje guardado en la BD");
            
        } catch (java.sql.SQLException e) {
            System.out.println("Error al guardar en BD: " + e.getMessage());
        }
    }
   
   public Personaje obtenerPersonaje(int indice){
       return this.listaPersonajes.get(indice);
   }
} 

File: python_app/controller/juego_controller.py
from db.conexion import conectar
from model.personaje import Guerrero, Mago, Arquero

class JuegoController:  
    def __init__(self):
        self.lista_personajes = []

    def agregar_personaje(self, personaje):
        self.lista_personajes.append(personaje)
        
        # Determinar el tipo y el atributo especial
        tipo = ""
        atributo = 0
        if isinstance(personaje, Guerrero):
            tipo = "Guerrero"
            atributo = personaje.ataque_espada
        elif isinstance(personaje, Mago):
            tipo = "Mago"
            atributo = personaje.hechizo_vida
        elif isinstance(personaje, Arquero):
            tipo = "Arquero"
            atributo = personaje.veneno

        # Guardar en BD
        conexion = conectar()
        if conexion:
            cursor = conexion.cursor()
            sql = "INSERT INTO personajes (nombre, nivel, vida, tipo, atributo_especial) VALUES (%s, %s, %s, %s, %s)"
            cursor.execute(sql, (personaje.nombre, personaje.nivel, personaje.vida, tipo, atributo))
            conexion.commit()
            cursor.close()
            conexion.close()
            print("¡Personaje guardado en la Base de Datos! 💾")